packages/pwa/app/commerce-api/contexts.js

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React, {createContext, useContext, useReducer} from 'react'
+
+/**
+ * Provider and associated hook for accessing the Commerce API in React components.
+ */
+export const CommerceAPIContext = createContext()
+export const CommerceAPIProvider = CommerceAPIContext.Provider
+export const useCommerceAPI = () => useContext(CommerceAPIContext)
+
+/**
+ * There are a few sources of global state in the react retail storefront.
+ * Using React Context we implement a simple shared global state allowing
+ * you can update and use either state from anywhere in the application.
+ *
+ * If your global state needs require a more robust solution, these contexts can be
+ * replaced by a third party state management library of your choosing, such as MobX
+ * or Redux.
+ *
+ * To use these context's simply import them into the component requiring context
+ * like the below example:
+ *
+ * import React, {useContext} from 'react'
+ * import {BasketContext} from 'components/_app-config'
+ *
+ * export const Avatar = () => {
+ *    const {customer} = useContext(BasketContext)
+ *    return <div>{customer.name}</div>
+ * }
+ *
+ */
+
+/************ Basket ************/
+export const BasketContext = createContext()
+export const BasketProvider = BasketContext.Provider
+
+/************ Customer ************/
+export const CustomerContext = createContext()
+export const CustomerProvider = CustomerContext.Provider
+
+/************ Customer Product Lists ************/
+const CPLInitialValue = {
+    isInitialized: false,
```

Comment:
I'm on the fence about this `isInitialized` state, action and reducer. The current implementation is essentially a toggle with it's default value set to false. Ideally `isInitialized` is a derived value on whether or not we have loaded product lists. This state is something that can be set internally when a product lists are loaded and set back to false when reset. 

The only problem I see is that because the initial state for the productLists is an object you can't derive this information. I think it would be a good idea to look at this again.

Reply:
removed!